UI hangs when loading / rendering large payload dataset

### Describe the bug

When you have a request with a lot of responses the UI frezzes upon opening large data.

### Expected behavior

The UI should not freeze but show a loading icon or somthing

### Actual behavior

The UI freezes and becomes unusable

### How to Reproduce?

1. load this yaml file 
https://file.io/pQ4qP3zQLNhN

2. go to the service and open the operation. The UI will freeze for 4~sec. Of cause adding examples or making them bigger worsen the problem.

### Microcks version or git rev

1.8.x

### Install method (`docker-compose`, `helm chart`, `operator`, `docker-desktop extension`,...)

docker-compose

### Additional information

_No response_

@lbroudoux How do you start the angular dev env.
I am currenltly having problems with the dependencies.
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: microcks-ui@1.0.0
npm ERR! Found: @angular/common@8.2.14
npm ERR! node_modules/@angular/common
npm ERR!   @angular/common@"~8.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/common@"<8.0.0" from patternfly-ng@4.5.2
npm ERR! node_modules/patternfly-ng
npm ERR!   patternfly-ng@"^4.5.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/apoorva/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/apoorva/.npm/_logs/2023-11-17T10_50_34_350Z-debug-0.log
```

--- a/webapp/src/main/webapp/src/app/app.module.ts
+++ b/webapp/src/main/webapp/src/app/app.module.ts
@@ -82,7 +82,7 @@ import { ArtifactUploaderDialogComponent } from './pages/importers/_components/u
 import { HubPageComponent } from './pages/hub/hub.page';
 import { HubPackagePageComponent } from './pages/hub/package/package.page';
 import { HubAPIVersionPageComponent } from './pages/hub/package/apiVersion/apiVersion.page';
-
+import { ExchangesTabsetComponent } from './pages/services/{serviceId}/_components/exchanges-tabset/exchanges-tabset.component';
 import json from 'highlight.js/lib/languages/json';
 import xml from 'highlight.js/lib/languages/xml';
 import yaml from 'highlight.js/lib/languages/yaml';
@@ -113,16 +113,17 @@ export function configLoader(configService: ConfigService) {
     TestsPageComponent, TestCreatePageComponent, TestDetailPageComponent, AddToCIDialogComponent, TestRunnerPageComponent,
     InvocationsServicePageComponent, ImportersPageComponent, ServiceRefsDialogComponent, ImporterWizardComponent,
     ArtifactUploaderDialogComponent, AdminPageComponent, SecretsTabComponent, SnapshotsTabComponent, UsersTabComponent,
-    GroupsManagementDialogComponent, HubPageComponent, HubPackagePageComponent, HubAPIVersionPageComponent
+    GroupsManagementDialogComponent, HubPageComponent, HubPackagePageComponent, HubAPIVersionPageComponent,
+    ExchangesTabsetComponent
   ],
   imports: [
     BrowserModule, BrowserAnimationsModule, FormsModule, AppRoutingModule, HttpClientModule,
     BsDatepickerModule.forRoot(), BsDropdownModule.forRoot(), ModalModule.forRoot(), TabsModule.forRoot(), TooltipModule.forRoot(),
     HighlightModule, FileUploadModule,
-    AboutModalModule, 
+    AboutModalModule,
     CardModule, DonutChartModule, SparklineChartModule,
     ListModule, PaginationModule, ToolbarModule,
-    WizardModule, ToastNotificationListModule, 
+    WizardModule, ToastNotificationListModule,
   ],
   providers: [
     ConfigService, {

--- a/webapp/src/main/webapp/src/app/pages/services/{serviceId}/_components/exchanges-tabset/exchanges-tabset.component.html
+++ b/webapp/src/main/webapp/src/app/pages/services/{serviceId}/_components/exchanges-tabset/exchanges-tabset.component.html
@@ -1,0 +1,246 @@
+<tabset #tabs>
+    <tab *ngFor="let exchange of this.view.messagesMap[item.name]; index as i">
+      <ng-template tabHeading>
+        {{ getExchangeName(exchange) }}
+        <span class="learn-more-inline">
+          <i tooltip="Source artifact: {{ getExchangeSourceArtifact(exchange) }}" containerClass="tooltip-break" class="pficon pficon-resources-almost-empty" aria-hidden="true"></i>
+        </span>
+      </ng-template>
+
+      <div *ngIf="shouldRender(i)">
+        <div class="row" *ngIf="isEventTypeService()">
+          <div class="col-md-6">
+            <h5 class="subsection-label">Message</h5>
+            <dl class="dl-horizontal left">
+              <dt>Content Type:</dt>
+              <dd><code><small>{{ exchange.eventMessage.mediaType }}</small></code></dd>
+            </dl>
+            <pre *ngIf="exchange.eventMessage.content"><code [highlight]="prettyPrintIfJSON(exchange.eventMessage.content)" [languages]="hlLang"></code></pre>
+            <table class="table table-condensed" width="100%" *ngIf="exchange.eventMessage.headers">
+              <thead>
+                <tr>
+                  <th>Header name</th>
+                  <th>Values</th>
+                </tr>
+              </thead>
+              <tbody>
+                <tr *ngFor="let header of exchange.eventMessage.headers">
+                  <td>{{ header.name }}</td>
+                  <td style="max-width:200px; word-wrap:break-word"><span *ngFor="let v of header.values; let last = last">{{ v }}{{ last ? '':', '}}</span></td>
+                </tr>
+              </tbody>
+            </table>
+          </div>
+          <div class="col-md-6">
+            <h5 class="subsection-label">Bindings details</h5>
+            <span *ngIf="hasBinding(item, 'KAFKA')">
+              <dl class="dl-horizontal left">
+                <dt>Kafka topic:</dt>
+                <dd>
+                  <div class="input-group">
+                    <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'KAFKA') }}" disabled/>
+                    <span class="input-group-addon">
+                      <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
+                        (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'KAFKA'))"></i>
+                    </span>
+                  </div>
+                </dd>
+                <dt>Kafka Key type:</dt>
+                <dd>{{ getBindingProperty(item, 'KAFKA', 'keyType') }}</dd>
+              </dl>
+            </span>
+            <span *ngIf="hasBinding(item, 'MQTT')">
+              <dl class="dl-horizontal left">
+                <dt>MQTT topic:</dt>
+                <dd>
+                  <div class="input-group">
+                    <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'MQTT') }}" disabled/>
+                    <span class="input-group-addon">
+                      <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
+                        (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'MQTT'))"></i>
+                    </span>
+                  </div>
+                </dd>
+                <dt>MQTT QoS:</dt>
+                <dd>{{ getBindingProperty(item, 'MQTT', 'qoS') }}</dd>
+                <dt>MQTT retain:</dt>
+                <dd>{{ getBindingProperty(item, 'MQTT', 'persistent') }}</dd>
+              </dl>
+            </span>
+            <span *ngIf="hasBinding(item, 'NATS')">
+              <dl class="dl-horizontal left">
+                <dt>NATS topic:</dt>
+                <dd>
+                  <div class="input-group">
+                    <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'NATS') }}" disabled/>
+                    <span class="input-group-addon">
+                      <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
+                        (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'NATS'))"></i>
+                    </span>
+                  </div>
+                </dd>
+              </dl>
+            </span>
+            <span *ngIf="hasBinding(item, 'WS')">
+              <dl class="dl-horizontal left">
+                <dt>WebSocket endpoint:</dt>
+                <dd>
+                  <div class="input-group">
+                    <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'WS') }}" disabled/>
+                    <span class="input-group-addon">
+                      <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
+                         (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'WS'))"></i>
+                    </span>
+                  </div>
+                </dd>
+                <dt>WS method:</dt>
+                <dd>{{ getBindingProperty(item, 'WS', 'method') }}</dd>
+              </dl>
+            </span>
+            <span *ngIf="hasBinding(item, 'AMQP')">
+              <dl class="dl-horizontal left">
+                <dt>AMQP exchange:</dt>
+                <dd>
+                  <div class="input-group">
+                    <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'AMQP') }}" disabled/>
+                    <span class="input-group-addon">
+                      <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
+                         (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'AMQP'))"></i>
+                    </span>
+                  </div>
+                </dd>
+                <dt>AMQP exchange type:</dt>
+                <dd>{{ getBindingProperty(item, 'AMQP', 'destinationType') }}</dd>
+              </dl>
+            </span>
+            <span *ngIf="hasBinding(item, 'GOOGLEPUBSUB')">
+              <dl class="dl-horizontal left">
+                <dt>Google PubSub topic:</dt>
+                <dd>
+                  <div class="input-group">
+                    <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'GOOGLEPUBSUB') }}" disabled/>
+                    <span class="input-group-addon">
+                      <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
+                         (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'GOOGLEPUBSUB'))"></i>
+                    </span>
+                  </div>
+                </dd>
+              </dl>
+            </span>
+            <span *ngIf="hasBinding(item, 'SQS')">
+              <dl class="dl-horizontal left">
+                <dt>AWS SQS Queue:</dt>
+                <dd>
+                  <div class="input-group">
+                    <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'SQS') }}" disabled/>
+                    <span class="input-group-addon">
+                      <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
+                         (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'SQS'))"></i>
+                    </span>
+                  </div>
+                </dd>
+                <dt>AWS SQS persistent:</dt>
+                <dd>{{ getBindingProperty(item, 'SQS', 'persistent') }}</dd>
+              </dl>
+            </span>
+            <span *ngIf="hasBinding(item, 'SNS')">
+              <dl class="dl-horizontal left">
+                <dt>AWS SNS Topic:</dt>
+                <dd>
+                  <div class="input-group">
+                    <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'SNS') }}" disabled/>
+                    <span class="input-group-addon">
+                      <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
+                         (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'SNS'))"></i>
+                    </span>
+                  </div>
+                </dd>
+              </dl>
+            </span>
+            <span *ngIf="hasBinding(item, 'AMQP1')">
+              <dl class="dl-horizontal left">
+                <dt>AMQP destination:</dt>
+                <dd>
+                  <div class="input-group">
+                    <input type="text" class="form-control" value="topic/UsersignedupAPI_0.1.0_user-signedup" disabled/>
+                    <span class="input-group-addon">
+                      <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
+                        (click)="copyToClipboard('topic/UsersignedupAPI_0.1.0_user-signedup')"></i>
+                    </span>
+                  </div>
+                </dd>
+              </dl>
+            </span>
+          </div>
+        </div>
+        <div class="row" *ngIf="!isEventTypeService()">
+          <div class="col-md-6">
+            <h5 class="subsection-label">Request</h5>
+            <dl>
+              <dt>Mock URL</dt>
+              <dd>
+                <div class="input-group">
+                  <input type="text" class="form-control" value="{{ formatMockUrl(item, exchange.response.dispatchCriteria) }}" disabled/>
+                  <span class="input-group-addon" style="border-left: 0;">
+                    <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
+                      (click)="copyToClipboard(formatMockUrl(item, exchange.response.dispatchCriteria))"></i>
+                  </span>
+                  <span class="input-group-addon" *ngIf="view.service.type != 'GRPC'">
+                    <i class="fa fa-terminal" tooltip="Curl to Clipboard" placement="top"
+                      (click)="copyToClipboard(formatCurlCmd(item, exchange))"></i>
+                  </span>
+                </div>
+              </dd>
+            </dl>
+            <pre *ngIf="exchange.request.content"><code [highlight]="formatRequestContent(exchange.request.content)" [languages]="hlLang"></code></pre>
+            <dl *ngIf="view.service.type === 'GRAPHQL'">
+              <dt>Variables</dt>
+              <dd>
+                <pre><code [highlight]="formatGraphQLVariables(exchange.request.content)" [languages]="hlLang"></code></pre>
+              </dd>
+            </dl>
+
+            <table class="table table-condensed" width="100%" *ngIf="exchange.request.headers">
+              <thead>
+                <tr>
+                  <th>Header name</th>
+                  <th>Values</th>
+                </tr>
+              </thead>
+              <tbody>
+                <tr *ngFor="let header of exchange.request.headers">
+                  <td>{{ header.name }}</td>
+                  <td style="max-width:200px; word-wrap:break-word"><span *ngFor="let v of header.values; let last = last">{{ v }}{{ last ? '':', '}}</span></td>
+                </tr>
+              </tbody>
+            </table>
+          </div>
+          <div class="col-md-6">
+            <h5 class="subsection-label">Response</h5>
+            <dl class="dl-horizontal left" *ngIf="view.service.type === 'REST' || view.service.type === 'GRAPHQL'">
+              <dt>Response Code and Type:</dt>
+              <dd>
+                <code><small>{{ exchange.response.status }}: {{exchange.response.mediaType}}</small></code>
+              </dd>
+            </dl>
+            <pre *ngIf="exchange.response.content"><code [highlight]="prettyPrintIfJSON(exchange.response.content)" [languages]="hlLang"></code></pre>
+            <table class="table table-condensed" width="100%" *ngIf="exchange.response.headers">
+              <thead>
+                <tr>
+                  <th>Header name</th>
+                  <th>Values</th>
+                </tr>
+              </thead>
+              <tbody>
+                <tr *ngFor="let header of exchange.response.headers">
+                  <td>{{ header.name }}</td>
+                  <td style="max-width:200px; word-wrap:break-word"><span *ngFor="let v of header.values; let last = last">{{ v }}{{ last ? '':', '}}</span></td>
+                </tr>
+              </tbody>
+            </table>
+          </div>
+        </div>
+
+      </div>
+    </tab>
+  </tabset>

--- a/webapp/src/main/webapp/src/app/pages/services/{serviceId}/_components/exchanges-tabset/exchanges-tabset.component.ts
+++ b/webapp/src/main/webapp/src/app/pages/services/{serviceId}/_components/exchanges-tabset/exchanges-tabset.component.ts
@@ -1,0 +1,62 @@
+/*
+ * Copyright The Microcks Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, ViewChild, Input } from '@angular/core';
+
+
+import { NotificationService, NotificationType } from 'patternfly-ng/notification';
+import { TabsetComponent } from 'ngx-bootstrap/tabs';
+import { ServiceView, Operation, ServiceType, Exchange, UnidirectionalEvent, RequestResponsePair, ParameterConstraint, EventMessage } from 'src/app/models/service.model';
+import { ConfigService } from 'src/app/services/config.service';
+
+
+@Component({
+  selector: 'app-exchanges-tabset',
+  templateUrl: './exchanges-tabset.component.html',
+})
+export class ExchangesTabsetComponent{
+
+  readonly hlLang: string[] = ['json', 'xml', 'yaml'];
+  @ViewChild('tabs', { static: true }) tabs?: TabsetComponent;
+  @Input() public item: Operation;
+  @Input() public view: ServiceView;
+  @Input() public resolvedServiceView: ServiceView;
+  @Input() public notificationService: NotificationService;
+  @Input() public isEventTypeService: Function;
+  @Input() public getExchangeName: Function;
+  @Input() public getExchangeSourceArtifact: Function;
+  @Input() public hasBinding: Function;
+  @Input() public getBindingProperty: Function;
+  @Input() public formatMockUrl: Function;
+  @Input() public formatAsyncDestination: Function;
+  @Input() public formatRequestContent: Function;
+  @Input() public formatGraphQLVariables: Function;
+  @Input() public prettyPrintIfJSON: Function;
+  @Input() public formatCurlCmd: Function;
+  @Input() public copyToClipboard: Function;
+  @Input() public encodeUrl: Function;
+  @Input() public removeVerbInUrl: Function;
+  @Input() public asyncAPIFeatureEndpoint: Function;
+  @Input() public config: ConfigService;
+
+
+
+
+  public shouldRender(index: number){
+      const activeTab = this.tabs.tabs.filter((tab) =>  tab.active )[0]
+      return index == this.tabs.tabs.indexOf(activeTab);
+  }
+
+}

--- a/webapp/src/main/webapp/src/app/pages/services/{serviceId}/service-detail.page.html
+++ b/webapp/src/main/webapp/src/app/pages/services/{serviceId}/service-detail.page.html
@@ -1,5 +1,5 @@
 <div class="form-group pull-right">
-  <pfng-toast-notification-list [notifications]="notifications" 
+  <pfng-toast-notification-list [notifications]="notifications"
       (onCloseSelect)="handleCloseNotification($event)" [showClose]="true">
   </pfng-toast-notification-list>
 </div>
@@ -137,7 +137,7 @@
         (onActionSelect)="handleAction($event, null)"
         (onDblClick)="handleDblClick($event)"
         (onSelectionChange)="handleSelectionChange($event)">
-      
+
         <ng-template #operationTemplate let-item="item" let-index="index">
           <div class="list-pf-left">
           </div>
@@ -163,7 +163,7 @@
                 <strong>  {{ view.messagesMap[item.name].length }}</strong> sample(s)
               </div>
             </div>
-            
+
             <div class="list-view-pf-actions">
               <div class="dropdown pull-right dropdown-kebab-pf" *ngIf="allowOperationsPropertiesEdit()">
                 <button class="btn btn-link dropdown-toggle" type="button" id="dropdownKebabRight9" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
@@ -189,7 +189,7 @@
               </span>
               <span *ngIf="!isEventTypeService()">
                 <dt>Dispatcher:</dt>
-                <dd><code>{{ item.dispatcher }}</code> &nbsp; 
+                <dd><code>{{ item.dispatcher }}</code> &nbsp;
                   <span class="learn-more-inline">
                     <a href="https://microcks.io/documentation/using/advanced/dispatching/" target="_blank">Learn More
                       <i class="fa fa-external-link" aria-hidden="true"></i>
@@ -207,7 +207,7 @@
                 </dd>
                 <dt>Available bindings:</dt>
                 <dd>
-                  {{ getBindingsList(item) }} &nbsp; 
+                  {{ getBindingsList(item) }} &nbsp;
                   <span class="learn-more-inline">
                     <a href="https://microcks.io/documentation/using/asyncapi/#bindings" target="_blank">Learn More
                       <i class="fa fa-external-link" aria-hidden="true"></i>
@@ -278,7 +278,7 @@
               <div class="input-group">
                 <input type="text" class="form-control" value="{{ formatMockUrl(item, null) }}" disabled/>
                 <span class="input-group-addon">
-                  <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top" 
+                  <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
                     (click)="copyToClipboard(formatMockUrl(item, null))"></i>
                 </span>
               </div>
@@ -310,249 +310,28 @@
           </div>
 
           <h4 class="section-label" *ngIf="view.service.type != 'GENERIC_REST'">Mocks</h4>
-          <tabset>
-            <tab *ngFor="let exchange of view.messagesMap[item.name]">
-              <ng-template tabHeading>
-                {{ getExchangeName(exchange) }}
-                <span class="learn-more-inline">
-                  <i tooltip="Source artifact: {{ getExchangeSourceArtifact(exchange) }}" containerClass="tooltip-break" class="pficon pficon-resources-almost-empty" aria-hidden="true"></i>
-                </span>
-              </ng-template>
-
-              <div class="row" *ngIf="isEventTypeService()">
-                <div class="col-md-6">
-                  <h5 class="subsection-label">Message</h5>
-                  <dl class="dl-horizontal left">
-                    <dt>Content Type:</dt>
-                    <dd><code><small>{{ exchange.eventMessage.mediaType }}</small></code></dd>
-                  </dl>
-                  <pre *ngIf="exchange.eventMessage.content"><code [highlight]="prettyPrintIfJSON(exchange.eventMessage.content)" [languages]="hlLang"></code></pre>
-                  <table class="table table-condensed" width="100%" *ngIf="exchange.eventMessage.headers">
-                    <thead>
-                      <tr>
-                        <th>Header name</th>
-                        <th>Values</th>
-                      </tr>
-                    </thead>
-                    <tbody>
-                      <tr *ngFor="let header of exchange.eventMessage.headers">
-                        <td>{{ header.name }}</td>
-                        <td style="max-width:200px; word-wrap:break-word"><span *ngFor="let v of header.values; let last = last">{{ v }}{{ last ? '':', '}}</span></td>
-                      </tr>
-                    </tbody>
-                  </table>
-                </div>
-                <div class="col-md-6">
-                  <h5 class="subsection-label">Bindings details</h5>
-                  <span *ngIf="hasBinding(item, 'KAFKA')">
-                    <dl class="dl-horizontal left">
-                      <dt>Kafka topic:</dt>
-                      <dd>
-                        <div class="input-group">
-                          <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'KAFKA') }}" disabled/>
-                          <span class="input-group-addon">
-                            <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top" 
-                              (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'KAFKA'))"></i>
-                          </span>
-                        </div>
-                      </dd>
-                      <dt>Kafka Key type:</dt>
-                      <dd>{{ getBindingProperty(item, 'KAFKA', 'keyType') }}</dd>
-                    </dl>
-                  </span>
-                  <span *ngIf="hasBinding(item, 'MQTT')">
-                    <dl class="dl-horizontal left">
-                      <dt>MQTT topic:</dt>
-                      <dd>
-                        <div class="input-group">
-                          <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'MQTT') }}" disabled/>
-                          <span class="input-group-addon">
-                            <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top" 
-                              (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'MQTT'))"></i>
-                          </span>
-                        </div>
-                      </dd>
-                      <dt>MQTT QoS:</dt>
-                      <dd>{{ getBindingProperty(item, 'MQTT', 'qoS') }}</dd>
-                      <dt>MQTT retain:</dt>
-                      <dd>{{ getBindingProperty(item, 'MQTT', 'persistent') }}</dd>
-                    </dl>
-                  </span>
-                  <span *ngIf="hasBinding(item, 'NATS')">
-                    <dl class="dl-horizontal left">
-                      <dt>NATS topic:</dt>
-                      <dd>
-                        <div class="input-group">
-                          <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'NATS') }}" disabled/>
-                          <span class="input-group-addon">
-                            <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top" 
-                              (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'NATS'))"></i>
-                          </span>
-                        </div>
-                      </dd>
-                    </dl>
-                  </span>
-                  <span *ngIf="hasBinding(item, 'WS')">
-                    <dl class="dl-horizontal left">
-                      <dt>WebSocket endpoint:</dt>
-                      <dd>
-                        <div class="input-group">
-                          <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'WS') }}" disabled/>
-                          <span class="input-group-addon">
-                            <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
-                               (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'WS'))"></i>
-                          </span>
-                        </div>
-                      </dd>
-                      <dt>WS method:</dt>
-                      <dd>{{ getBindingProperty(item, 'WS', 'method') }}</dd>
-                    </dl>
-                  </span>
-                  <span *ngIf="hasBinding(item, 'AMQP')">
-                    <dl class="dl-horizontal left">
-                      <dt>AMQP exchange:</dt>
-                      <dd>
-                        <div class="input-group">
-                          <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'AMQP') }}" disabled/>
-                          <span class="input-group-addon">
-                            <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
-                               (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'AMQP'))"></i>
-                          </span>
-                        </div>
-                      </dd>
-                      <dt>AMQP exchange type:</dt>
-                      <dd>{{ getBindingProperty(item, 'AMQP', 'destinationType') }}</dd>
-                    </dl>
-                  </span>
-                  <span *ngIf="hasBinding(item, 'GOOGLEPUBSUB')">
-                    <dl class="dl-horizontal left">
-                      <dt>Google PubSub topic:</dt>
-                      <dd>
-                        <div class="input-group">
-                          <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'GOOGLEPUBSUB') }}" disabled/>
-                          <span class="input-group-addon">
-                            <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
-                               (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'GOOGLEPUBSUB'))"></i>
-                          </span>
-                        </div>
-                      </dd>
-                    </dl>
-                  </span>
-                  <span *ngIf="hasBinding(item, 'SQS')">
-                    <dl class="dl-horizontal left">
-                      <dt>AWS SQS Queue:</dt>
-                      <dd>
-                        <div class="input-group">
-                          <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'SQS') }}" disabled/>
-                          <span class="input-group-addon">
-                            <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
-                               (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'SQS'))"></i>
-                          </span>
-                        </div>
-                      </dd>
-                      <dt>AWS SQS persistent:</dt>
-                      <dd>{{ getBindingProperty(item, 'SQS', 'persistent') }}</dd>
-                    </dl>
-                  </span>
-                  <span *ngIf="hasBinding(item, 'SNS')">
-                    <dl class="dl-horizontal left">
-                      <dt>AWS SNS Topic:</dt>
-                      <dd>
-                        <div class="input-group">
-                          <input type="text" class="form-control" value="{{ formatAsyncDestination(item, exchange.eventMessage, 'SNS') }}" disabled/>
-                          <span class="input-group-addon">
-                            <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top"
-                               (click)="copyToClipboard(formatAsyncDestination(item, exchange.eventMessage, 'SNS'))"></i>
-                          </span>
-                        </div>
-                      </dd>
-                    </dl>
-                  </span>
-                  <span *ngIf="hasBinding(item, 'AMQP1')">
-                    <dl class="dl-horizontal left">
-                      <dt>AMQP destination:</dt>
-                      <dd>
-                        <div class="input-group">
-                          <input type="text" class="form-control" value="topic/UsersignedupAPI_0.1.0_user-signedup" disabled/>
-                          <span class="input-group-addon">
-                            <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top" 
-                              (click)="copyToClipboard('topic/UsersignedupAPI_0.1.0_user-signedup')"></i>
-                          </span>
-                        </div>
-                      </dd>
-                    </dl>
-                  </span>
-                </div>
-              </div>
-              <div class="row" *ngIf="!isEventTypeService()">
-                <div class="col-md-6">
-                  <h5 class="subsection-label">Request</h5>
-                  <dl>
-                    <dt>Mock URL</dt>
-                    <dd>
-                      <div class="input-group">
-                        <input type="text" class="form-control" value="{{ formatMockUrl(item, exchange.response.dispatchCriteria) }}" disabled/>
-                        <span class="input-group-addon" style="border-left: 0;">
-                          <i class="fa fa-paste" tooltip="Copy to Clipboard" placement="top" 
-                            (click)="copyToClipboard(formatMockUrl(item, exchange.response.dispatchCriteria))"></i>
-                        </span>
-                        <span class="input-group-addon" *ngIf="view.service.type != 'GRPC'">
-                          <i class="fa fa-terminal" tooltip="Curl to Clipboard" placement="top"
-                            (click)="copyToClipboard(formatCurlCmd(item, exchange))"></i>
-                        </span>
-                      </div>
-                    </dd>
-                  </dl>
-                  <pre *ngIf="exchange.request.content"><code [highlight]="formatRequestContent(exchange.request.content)" [languages]="hlLang"></code></pre>
-                  <dl *ngIf="view.service.type === 'GRAPHQL'">
-                    <dt>Variables</dt>
-                    <dd>
-                      <pre><code [highlight]="formatGraphQLVariables(exchange.request.content)" [languages]="hlLang"></code></pre>
-                    </dd>
-                  </dl>
-
-                  <table class="table table-condensed" width="100%" *ngIf="exchange.request.headers">
-                    <thead>
-                      <tr>
-                        <th>Header name</th>
-                        <th>Values</th>
-                      </tr>
-                    </thead>
-                    <tbody>
-                      <tr *ngFor="let header of exchange.request.headers">
-                        <td>{{ header.name }}</td>
-                        <td style="max-width:200px; word-wrap:break-word"><span *ngFor="let v of header.values; let last = last">{{ v }}{{ last ? '':', '}}</span></td>
-                      </tr>
-                    </tbody>
-                  </table>
-                </div>
-                <div class="col-md-6">
-                  <h5 class="subsection-label">Response</h5>
-                  <dl class="dl-horizontal left" *ngIf="view.service.type === 'REST' || view.service.type === 'GRAPHQL'">
-                    <dt>Response Code and Type:</dt>
-                    <dd>
-                      <code><small>{{ exchange.response.status }}: {{exchange.response.mediaType}}</small></code>
-                    </dd>
-                  </dl>
-                  <pre *ngIf="exchange.response.content"><code [highlight]="prettyPrintIfJSON(exchange.response.content)" [languages]="hlLang"></code></pre>
-                  <table class="table table-condensed" width="100%" *ngIf="exchange.response.headers">
-                    <thead>
-                      <tr>
-                        <th>Header name</th>
-                        <th>Values</th>
-                      </tr>
-                    </thead>
-                    <tbody>
-                      <tr *ngFor="let header of exchange.response.headers">
-                        <td>{{ header.name }}</td>
-                        <td style="max-width:200px; word-wrap:break-word"><span *ngFor="let v of header.values; let last = last">{{ v }}{{ last ? '':', '}}</span></td>
-                      </tr>
-                    </tbody>
-                  </table>
-                </div>
-              </div>
-            </tab>
-          </tabset>
+          <app-exchanges-tabset
+          [item]="item"
+          [view]="view"
+          [resolvedServiceView]="resolvedServiceView"
+          [notificationService]="notificationService"
+          [isEventTypeService]="isEventTypeService"
+          [getExchangeName]="getExchangeName"
+          [getExchangeSourceArtifact]="getExchangeSourceArtifact"
+          [hasBinding]="hasBinding"
+          [getBindingProperty]="getBindingProperty"
+          [formatMockUrl]="formatMockUrl"
+          [formatAsyncDestination]="formatAsyncDestination"
+          [formatRequestContent]="formatRequestContent"
+          [formatGraphQLVariables]="formatGraphQLVariables"
+          [prettyPrintIfJSON]="prettyPrintIfJSON"
+          [formatCurlCmd]="formatCurlCmd"
+          [copyToClipboard]="copyToClipboard"
+          [removeVerbInUrl]="removeVerbInUrl"
+          [encodeUrl]="encodeUrl"
+          [asyncAPIFeatureEndpoint]="asyncAPIFeatureEndpoint"
+          [config]="config"
+          ></app-exchanges-tabset>
         </ng-template>
 
     </pfng-list>

